txclib/wizard.py

```diff
@@ -0,0 +1,190 @@
+import os
+import readline
+
+from slugify import slugify
+from txclib import messages
+from txclib import utils
+from txclib.api import Api
+from txclib.project import Project
+from txclib.log import logger
+from six.moves import input
+
+
+COLOR = "CYAN"
+
+readline.set_completer_delims(' \t\n;')
+readline.parse_and_bind("tab: complete")
+
+
+def validate_source_file(path):
+    return os.path.isfile(os.path.abspath(path))
+
+
+def validate_expression(expression):
+    return '<lang>' in expression
+
+
+def validate_int(choice, length):
+    try:
+        choice = int(choice)
+    except ValueError:
+        return False
+    return 0 < choice <= length
+
+
+def choice_prompt(l, key):
+    """
+    l: A list of tuples (key, display_value) with the valid choices
+    key: one of 'formats', 'organizations', 'projects'
+    returns the key of the selected choice
+    """
+    a = "\n".join(["  {}. {}".format(i+1, f[1])
+                   for i, f in enumerate(l)])
+    a = a + "\n"
+    print(a)
+
+    choice = ''
+    first_time = True
+    r = '1' if len(l) == 1 else '1-{}'.format(len(l))
+
+    while not validate_int(choice, len(l)):
+        if not first_time:
+            print(messages.TEXTS[key]["error"])
+        choice = input(utils.color_text(
+            messages.TEXTS[key]['message'].format(r=r), COLOR))
+        first_time = False
+    return l[int(choice) - 1][0]
+
+
+def input_prompt(key, validation_method):
+    user_input = ''
+    first_time = True
+    while not validation_method(user_input):
+        if not first_time:
+            print(messages.TEXTS[key]['error'])
+
+        user_input = input(
+            utils.color_text(messages.TEXTS[key]['message'], COLOR))
+        first_time = False
+    return user_input
+
+
+class Wizard(object):
+
+    def __init__(self, path_to_tx):
+        p = Project(path_to_tx)
+        self.host = p.config.get('main', 'host')
+        username, token_or_password = p.getset_host_credentials(
+            self.host, only_token=True)
+
+        self.api = Api(username=username, password=token_or_password,
+                       host=self.host, path_to_tx=p.txrc_file)
+
+    def get_organizations(self):
+        try:
+            organizations = self.api.get('organizations')
+        except Exception as e:
+            logger.error(e)
+            exit(1)
```

Comment:
you probably want `sys.exit` here. See https://docs.python.org/2.7/library/constants.html#constants-added-by-the-site-module

Reply:
Also, generally speaking it seems to me you might want to do something like this in these methods:
```
try:
    ...
except Exception:
    logger.exception('requesting organizations')
    raise
```

Then in `.run()` you could catch and call `sys.exit(1)`

Reply:
Actually I removed the `exit()` altogether and raised the Exception instead. sys.exit(1) will be called anyway from the global error handling we have in main.